### PR TITLE
Fixed hide-corners bug

### LINKED ISSUE #139: Panel corners black after toggling overview
When toggling overview on gnome 40 the panel corners become black if hide corners is enabled, and become semi-transparent otherwise. I'm on arch using nvidia proprietary drivers on xorg. Not sure if any of that matters :) great extension btw!!

### PR DESCRIPTION
**Pull request by ajbucci**
Hey, this is my first pull request so apologies if I haven't done this quite right! This should fix the issue "Panel corners black after toggling overview"

Fixes #139

### CODE REVIEW

**Comment by ajbucci** [comment]
This is only a partial fix. The black corners will return if the background custom opacities are adjusted, though re-enabling the extension fixes it until the opacities are adjusted again. It may be due to how set_corner_color is working in theming.js?

**Comment by Bananaman** [comment]
@ewlsh @ajbucci Thanks for trying to solve this issue. It looks like the solution here might be over-engineered?

I posted a comment on the issue ticket, which mentions how to solve this problem with 1 line of CSS:

https://github.com/ewlsh/dynamic-panel-transparency/issues/139#issuecomment-827136784

File: dynamic-panel-transparency@rockon999.github.io/stylesheet.css [hunk 1]
```diff
@@ -20,3 +20,6 @@
   background-image: none;
 }
 
+/* Prevents overlapping of corners + top bar when using transparency */
+.panel-corner {
```
Comment (ewlsh) [inline dynamic-panel-transparency@rockon999.github.io/stylesheet.css:24]:
```suggestion
.dpt-panel-hide-corners {
```

File: dynamic-panel-transparency@rockon999.github.io/theming.js [hunk 2]
```diff
@@ -150,6 +150,31 @@ function has_icon_shadow() {
 function remove_text_shadow() {
     Panel.remove_style_class_name('dpt-panel-text-shadow');
 }
+/**
```
Comment (ewlsh) [inline dynamic-panel-transparency@rockon999.github.io/theming.js:153]: Remove `register_hide_corners` - you don't need a dynamic stylesheet for a static style.

File: dynamic-panel-transparency@rockon999.github.io/extension.js [hunk 3]
```diff
@@ -206,7 +221,7 @@ function initialize_settings() {
 
                 for (let i = Theming.stylesheets.length - 1; i >= 0; i--) {
                     let stylesheet = Theming.stylesheets[i];
-                    if (stylesheet.indexOf('background') !== -1 && stylesheet.indexOf('panel-') !== -1) {
+                    if (stylesheet.indexOf('background') !== -1 && stylesheet.indexOf('panel-') !== -1 && stylesheet.indexOf('panel-hide-corners') == -1) {
```
Comment (ewlsh) [inline dynamic-panel-transparency@rockon999.github.io/extension.js:224]: Revert.

File: dynamic-panel-transparency@rockon999.github.io/extension.js [hunk 4]
```diff
@@ -247,7 +262,7 @@ function initialize_settings() {
...
```
Comment (ewlsh) [inline dynamic-panel-transparency@rockon999.github.io/extension.js:265]: Revert.

File: dynamic-panel-transparency@rockon999.github.io/extension.js [hunk 5]
```diff
@@ -285,7 +300,7 @@ function initialize_settings() {
 
                 for (let i = Theming.stylesheets.length - 1; i >= 0; i--) {
                     let stylesheet = Theming.stylesheets[i];
-                    if (stylesheet.indexOf('background') !== -1 && stylesheet.indexOf('panel.dpt.css') !== -1) {
+                    if (stylesheet.indexOf('background') !== -1 && stylesheet.indexOf('panel.dpt.css') !== -1 && stylesheet.indexOf('panel-hide-corners') == -1) {
```
Comment (ewlsh) [inline dynamic-panel-transparency@rockon999.github.io/extension.js:303]: Revert.

File: dynamic-panel-transparency@rockon999.github.io/extension.js [hunk 6]
```diff
@@ -106,14 +106,25 @@ function modify_panel() {
...
     if (Settings.get_enable_text_color()) {
         if (text !== null) {
             Theming.set_text_color();
         } else {
             log('[Dynamic Panel Transparency] Failed to enabled text coloring.');
         }
     }
+
+    /* Register hide corners. */
+    let hide_corners = Theming.register_hide_corners();
```
Comment (ewlsh) [inline dynamic-panel-transparency@rockon999.github.io/extension.js:119]:
```suggestion
```

File: dynamic-panel-transparency@rockon999.github.io/extension.js [hunk 7]
```diff
@@ -106,14 +106,25 @@ function modify_panel() {
...
+    
     if (Settings.get_enable_text_color()) {
         if (text !== null) {
             Theming.set_text_color();
         } else {
             log('[Dynamic Panel Transparency] Failed to enabled text coloring.');
         }
     }
+
+    /* Register hide corners. */
```
Comment (ewlsh) [inline dynamic-panel-transparency@rockon999.github.io/extension.js:118]:
```suggestion
```

File: dynamic-panel-transparency@rockon999.github.io/extension.js [hunk 8]
```diff
@@ -106,14 +106,25 @@ function modify_panel() {
...
+    let hide_corners = Theming.register_hide_corners();
+
+    if (Settings.get_hide_corners()) {
+        if (hide_corners !== null) {
```
Comment (ewlsh) [inline dynamic-panel-transparency@rockon999.github.io/extension.js:122]: Remove conditional.

File: dynamic-panel-transparency@rockon999.github.io/stylesheet.css [hunk 9]
```diff
@@ -20,3 +20,6 @@
...
+  -panel-corner-border-width: 0px;}
```
Comment (ewlsh) [inline dynamic-panel-transparency@rockon999.github.io/stylesheet.css:25]:
```suggestion
  -panel-corner-border-width: 0px;
}

```

File: dynamic-panel-transparency@rockon999.github.io/extension.js [hunk 10]
```diff
@@ -106,14 +106,25 @@ function modify_panel() {
     /* Register text color styling. */
     let [text, icon, arrow] = Theming.register_text_color(Settings.get_text_color()); // eslint-disable-line no-unused-vars
     let [maximized_text, maximized_icon, maximized_arrow] = Theming.register_text_color(Settings.get_maximized_text_color(), 'maximized'); // eslint-disable-line no-unused-vars
-
+
```
Comment (ewlsh) [inline dynamic-panel-transparency@rockon999.github.io/extension.js:109]:
```suggestion

```

**Review by ewlsh — CHANGES_REQUESTED** [review]

**Comment by ewlsh** [comment]
@ajbucci I reviewed this and listed the necessary changes, I'll apply them myself and commit this weekend unless you'd like to.

**Comment by ajbucci** [comment]
@ewlsh Thanks for taking the time to review the changes! I'm a bit shot at the moment as my wife and I just had our first child this past week and are still adjusting. Go ahead and apply your changes/suggestions this weekend. Thanks again!

**Comment by ewlsh** [comment]
Changes were manually applied, thank you :wave:

**Pull request closed without merging**